library/std/src/sys/unix/ext/fs.rs

```diff
@@ -345,6 +345,33 @@ pub trait OpenOptionsExt {
     /// ```
     #[stable(feature = "open_options_ext", since = "1.10.0")]
     fn custom_flags(&mut self, flags: i32) -> &mut Self;
+
+    /// Get the flags of this OpenOptions as libc::c_int.
+    ///
+    /// This method allows the reuse of the OpenOptions as flags argument for `libc::open()`.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #![feature(rustc_private)]
+    /// extern crate libc;
+    /// use std::ffi::CString;
+    /// use std::fs::OpenOptions;
+    /// use std::os::unix::fs::OpenOptionsExt;
+    ///
+    /// # fn main() {
+    /// let mut options = OpenOptions::new();
+    /// options.write(true).read(true);
```

Comment:
```suggestion
    /// let options = OpenOptions::new().write(true).read(true);
```
We could just reuse `mut`.

Reply:
I've encountered the following error:
```
error[E0716]: temporary value dropped while borrowed
  --> src/sys/unix/ext/fs.rs:362:19
   |
10 | let mut options = OpenOptions::new().write(true).read(true);
   |                   ^^^^^^^^^^^^^^^^^^                       - temporary value is freed at the end of this statement
   |                   |
   |                   creates a temporary which is freed while still in use
11 | let file_name = CString::new("foo.txt").unwrap();
12 | let file = unsafe { libc::open(file_name.as_c_str().as_ptr(), options.as_flags().unwrap()) };
   |                                                               ------- borrow later used here
   |
   = note: consider using a `let` binding to create a longer lived value

```
Thus I would keep the old version, which is also similar to what can be found on the other function's docs if that's ok